samples/apps/copilot-chat-app/webapi/Services/AzureContentModerator.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI;
+using SemanticKernel.Service.CopilotChat.Options;
+
+namespace SemanticKernel.Service.Services;
+
+public record AnalysisResult(
+    [property: JsonPropertyName("category")] string Category,
+    [property: JsonPropertyName("riskLevel")] short RiskLevel
+);
+
+public record ImageContent([property: JsonPropertyName("content")] string Content);
+
+public record ImageAnalysisRequest(
+    [property: JsonPropertyName("image")] ImageContent Image,
+    [property: JsonPropertyName("categories")] List<string> Categories
+);
+
+public sealed class AzureContentModerator : IDisposable
+{
+    private const string HttpUserAgent = "Copilot Chat";
+
+    private readonly Uri _endpoint;
+    private readonly HttpClient _httpClient;
+    private readonly HttpClientHandler? _httpClientHandler;
+
+    /// <summary>
+    /// Options for the content moderator.
+    /// </summary>
+    private readonly ContentModerationOptions? _contentModerationOptions;
+
+    /// <summary>
+    /// Gets the options for the content moderator.
+    /// </summary>
+    public ContentModerationOptions? ContentModerationOptions => this._contentModerationOptions;
+
+    private static readonly List<string> s_categories = new()
+    {
+        "Hate",
```

Comment:
Those words would make for a fun t-shirt! 🤣

Reply:
hahah I'll pitch it to John